pkg/utils/host.go

```diff
@@ -0,0 +1,135 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+)
+
+const (
+	SriovConfBasePath     = "/etc/sriov-operator"
+	PfAppliedConfig       = SriovConfBasePath + "/pci"
+	HostSriovConfBasePath = "/host" + SriovConfBasePath
+	HostPfAppliedConfig   = HostSriovConfBasePath + "/pci"
+)
+
+type PfStatus struct {
+	NumVfs            int    `json:"numVfs"`
+	Mtu               int    `json:"mtu"`
+	LinkType          string `json:"linkType"`
+	EswitchMode       string `json:"eSwitchMode"`
+	ExternallyCreated bool   `json:"externallyCreated"`
+}
+
+// create the operator base folder on the host together with the pci folder to save the PF status objects as json files
+func CreateOperatorConfigFolderIfNeeded() error {
+	_, err := os.Stat(SriovConfBasePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = os.Mkdir(SriovConfBasePath, os.ModeDir)
+			if err != nil {
+				return fmt.Errorf("failed to create the sriov config folder on host in path %s: %v", SriovConfBasePath, err)
+			}
+		} else {
+			return fmt.Errorf("failed to check if the sriov config folder on host in path %s exist: %v", SriovConfBasePath, err)
+		}
+	}
+
+	_, err = os.Stat(PfAppliedConfig)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = os.Mkdir(PfAppliedConfig, os.ModeDir)
```

Comment:
you can call MkdirAll[1] on PfAppliedConfg

[1[ https://pkg.go.dev/os#MkdirAll